dcore/advection.py

```diff
@@ -122,3 +115,53 @@ def apply(self, x_in, x_out):
         dt = self.state.timestepping.dt
         self.solver.solve()
         x_out.assign(x_in + dt*self.dq)
+
+
+class DGAdvection(Advection):
+
+    def __init__(self, state):
+
+        super(DGAdvection, self).__init__(state)
+
+        dt = state.timestepping.dt
+
+        # by convention the last space is the DG space
+        DGSpace = state.V[-1]
```

Comment:
Could you not just have a list and put it in Form order. That would provide
a definitive reason why DG is space -1, and would provide places for the
other ones?

On Thu, 17 Mar 2016 at 16:34 jshipton <notifications@github.com> wrote:

> In dcore/advection.py
> <https://github.com/firedrakeproject/dcore/pull/23#discussion_r56534669>:
>
> > @@ -122,3 +115,53 @@ def apply(self, x_in, x_out):
> >          dt = self.state.timestepping.dt
> >          self.solver.solve()
> >          x_out.assign(x_in + dt*self.dq)
> > +
> > +
> > +class DGAdvection(Advection):
> > +
> > +    def __init__(self, state):
> > +
> > +        super(DGAdvection, self).__init__(state)
> > +
> > +        dt = state.timestepping.dt
> > +
> > +        # by convention the last space is the DG space
> > +        DGSpace = state.V[-1]
>
> I like the idea of a get_space method - much better than a list. I'm just
> hesitating because I don't know what to call the spaces… I don't want to
> name them after the field because that is different depending on the
> equations we are solving - for shallow water the DG field is the depth
> field, for vertical slice it is density (or pressure if you're
> incompressible). OK so I can call the DG space "DGspace" but what about the
> velocity space, which is RT if you're on quads but BD(F)M if you're on
> triangles… it's a bit inconsistent but I suppose that is always the
> velocity space so I can name that one after the field… maybe this ramble
> has helped me figure out an answer?!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/firedrakeproject/dcore/pull/23/files/4837fc777b71aa2914a450213721a3661068e0df#r56534669>
>